src/cartesian/Bar.tsx

```diff
@@ -738,8 +740,8 @@ export function computeBarRectangles({
       let value, x: number | null, y, width, height, background: Rectangle;
 
       if (stackedData) {
-        // we don't need to use dataStartIndex here, because stackedData is already sliced from the selector
```

Comment:
What is the situation here? The old comment seems to contradict the PR.

Reply:
The old comment was wrong. stackedData was never sliced in the selector (only displayedData was sliced at line 527). That's exactly why this fix is needed, we must use index + dataStartIndex to access the correct element in the full (non-sliced) stackedData array.